metadata-ingestion/src/datahub/ingestion/source/gc/soft_deleted_entity_cleanup.py

```diff
@@ -157,68 +186,127 @@ def delete_soft_deleted_entity(self, urn: str) -> None:
             ]["created"]["time"] < (retention_time * 1000):
                 logger.debug(f"Hard deleting {urn}")
                 self.delete_entity(urn)
+            else:
+                self._increment_retained_count()
+
+    def _print_report(self) -> None:
```

Comment:
does periodic reporting need to be managed for every report?
isn't that handled by the pipeline-report integration itself?

Reply:
That is done only if source returns with workunits. But this source does not return any workunit. So it never prints until the end.